packages/components/src/components/SquiggleChart.tsx

```diff
@@ -53,38 +52,66 @@ export interface SquiggleChartProps {
   /** Whether to show vega actions to the user, so they can copy the chart spec */
   distributionChartActions?: boolean;
   enableLocalSettings?: boolean;
-}
+} & (StandaloneExecutionProps | ProjectExecutionProps);
+
+// Props needed for a standalone execution
+type StandaloneExecutionProps = {
+  /** Project must be undefined */
+  project?: undefined;
+  /** Includes must be undefined */
+  includes?: undefined;
+  /** The amount of points returned to draw the distribution, not needed if using a project */
+  environment?: environment;
+};
 
+// Props needed when executing inside a project.
+type ProjectExecutionProps = {
+  /** environment must be undefined (we don't set it here, users can set the environment outside the execution) */
+  environment?: undefined;
+  /** The project that this execution is part of */
+  project: SqProject;
+  /** What other squiggle sources from the project to include. Default none */
+  includes?: string[];
+};
 const defaultOnChange = () => {};
 const defaultImports: JsImports = {};
 
 export const SquiggleChart: React.FC<SquiggleChartProps> = React.memo(
-  ({
-    code = "",
-    executionId = 0,
-    environment,
-    onChange = defaultOnChange, // defaultOnChange must be constant, don't move its definition here
-    height = 200,
-    jsImports = defaultImports,
-    showSummary = false,
-    width,
-    logX = false,
-    expY = false,
-    diagramStart = 0,
-    diagramStop = 10,
-    diagramCount = 20,
-    tickFormat,
-    minX,
-    maxX,
-    color,
-    title,
-    xAxisType = "number",
-    distributionChartActions,
-    enableLocalSettings = false,
-  }) => {
+  (props: SquiggleChartProps) => {
+    const {
+      executionId = 0,
+      onChange = defaultOnChange, // defaultOnChange must be constant, don't move its definition here
+      height = 200,
+      jsImports = defaultImports,
+      showSummary = false,
+      width,
+      logX = false,
+      expY = false,
+      diagramStart = 0,
+      diagramStop = 10,
+      diagramCount = 20,
+      tickFormat,
+      minX,
+      maxX,
+      color,
+      title,
+      xAxisType = "number",
+      distributionChartActions,
+      enableLocalSettings = false,
+      project = SqProject.create(),
+      code,
+      includes = [],
+    } = props;
+
+    const p = project ?? SqProject.create();
+    if (!project && props.environment) {
+      p.setEnvironment(props.environment);
+    }
```

Comment:
There's a potential problem here. Or several problems.

If you change the environment prop on a rerender, this won't cause reexecution, after you fix my `useMemo` note above.

And then, also, even if you did reexecute, ReducerProject won't rerun all of the included sources if they were executed before.
(This is a bug in `ReducerProject`; it should clean up the result from `ReducerProject_ProjectItem`s through `touchSource` or `touchRawParse` when `project.setEnvironment` is called.)

I guess at least the first part of this can be fixed by putting `project.getEnvironment()` in `useSquiggle`'s dependency array for `useMemo`?